src/bijectors/ordered.jl

```diff
@@ -88,3 +78,72 @@ end
 
 logabsdetjac(b::OrderedBijector, x::AbstractVector) = sum(@view(x[2:end]))
 logabsdetjac(b::OrderedBijector, x::AbstractMatrix) = vec(sum(@view(x[2:end, :]); dims=1))
+
+# Need a custom distribution type to handle this properly.
+struct OrderedDistribution{B,D<:ContinuousMultivariateDistribution} <:
+       ContinuousMultivariateDistribution
+    dist::D
+    bijector::B
+end
+
+"""
+    ordered(d::Distribution)
+
+Return a `Distribution` whose support are ordered vectors, i.e., vectors with increasingly ordered elements.
+
+This transformation is currently only supported for otherwise unconstrained distributions.
+"""
+function ordered(d::ContinuousMultivariateDistribution)
+    # We're good if the map from unconstrained (in which we apply the ordered bijector)
+    # to constrained is monotonically increasing, i.e. order-preserving. In that case,
+    # we can form the ordered transformation as `binv ∘ OrderedBijector() ∘ b`.
+    # Similarly, if we're working with monotonically decreasing maps, we can do the same
+    # but with the addition of a sign flip before and after the ordered bijector.
+    b = bijector(d)
+    binv = inverse(b)
+    if is_monotonically_decreasing(binv)
+        ordered_b = binv ∘ SignFlip() ∘ OrderedBijector() ∘ SignFlip() ∘ b
+    elseif is_monotonically_increasing(binv)
+        ordered_b = binv ∘ OrderedBijector() ∘ b
+    else
+        throw(ArgumentError("ordered transform is currently not supported for $d."))
+    end
+
+    return OrderedDistribution(d, inverse(ordered_b))
+end
+
+bijector(d::OrderedDistribution) = d.bijector
+
+function Distributions._logpdf(d::OrderedDistribution, x::AbstractVector{<:Real})
+    issorted(x) || return -eltype(d)(Inf)
+    return Distributions.logpdf(d.dist, x)
+end
+Base.length(d::OrderedDistribution) = length(d.dist)
+
+"""
+    marginal_dist(d::MultivariateDistribution, i::Int)
+
+Return the marginal distribution of the `i`-th component of the multivariate
+distribution `d`.
+"""
+marginal_dist(d::Distributions.Product, i::Int) = d.v[i]
+function marginal_dist(d::AbstractMvNormal, i::Int)
+    return Normal(mean(d)[i], sqrt(var(d)[i]))
+end
+
+function Distributions._rand!(
+    rng::AbstractRNG, d::OrderedDistribution, x::AbstractVector{<:Real}
+)
+    # Rejection sampling.
+    x[1] = rand(rng, marginal_dist(d.dist, 1))
+    for i in 2:length(x)
+        x_prev = x[i - 1]
+        d_i = marginal_dist(d.dist, i)
+        success = false
+        while !success
+            x[i] = rand(rng, d_i)
+            success |= x[i] >= x_prev
+        end
+    end
+    return x
```

Comment:
This implementation seems to assume a `Product` structure. For `MvNormal` it would certainly do the wrong thing.

Why not just use this for the generic case:
```suggestion
    # Rejection sampling.
    while true
        x = rand(rng, d.dist)
        issorted(x) && return x
    end
```

And then specialize for `ProductDistribution`?